### PR TITLE
Updating docstring of a class according to PEP 257 in a file

### PR DESCRIPTION
**Pull request by ashish-hacker**
link to the concerned issue #4414 

Depending on what your PR does, here are a few things you might want to address in the description:
+ [x] what are the (breaking) changes that this PR makes?
- This PR is about updating docstrings to PEP 257.
+ [x] important background, or details about the implementation
- Used PEP 257 
+ [x] are the changes—especially new features—covered by tests and docstrings?
- Used PEP 257 for docstrings instead of numpy docstrings
+ [x] [linting/style checks have been run](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style)
- Done style checking as mentioned in [this](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style)
+ [ ] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples)
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

[link to the changed docstrings](https://github.com/pymc-devs/pymc3/blob/2d3ec8f3afcbaa9dcb8401af5fff6932ad7fd158/pymc3/step_methods/metropolis.py#L97)

### CODE REVIEW

**Comment by ashish-hacker** [comment]
@michaelosthege Kindly review it.

**Review by OriolAbril — COMMENTED** [review]
I think the numpydoc formatting should be kept and change only where the docstring is, I don't think napoleon supports that format.

Can you generate docs locally? I amnactually not sure if there is any documentation on how to do so anywhere

**Comment by michaelosthege** [comment]
Yes, the formatting of the `Parameters` section needs to remain the same.
And the first line in the `__init__` docstring should be something like "Create an instance of a Metropolis stepper"

File: pymc3/step_methods/metropolis.py [hunk 1]
```diff
@@ -144,6 +143,19 @@ def __init__(
         mode=None,
         **kwargs
     ):
+        """Create an instance of a Metropolis stepper
+
+        Arguments:
+        vars -- List of variables for sampler (default None)
```
Comment (twiecki) [inline pymc3/step_methods/metropolis.py]: This looks very different from all our other doc strings which follow the numpy style guide, can you make it adhere to that?

Comment (ashish-hacker) [reply pymc3/step_methods/metropolis.py]: Oh Sorry! My bad. I will fix it right away

Comment (ashish-hacker) [reply pymc3/step_methods/metropolis.py]: Done @twiecki : )

File: pymc3/step_methods/metropolis.py [hunk 2]
```diff
@@ -144,6 +122,28 @@ def __init__(
...
+            S (and n). Defaults to normal.
+        scaling: scalar or array
+            Initial scale factor for proposal. Defaults to 1.
+        tune: bool
+            Flag for tuning. Defaults to True.
+        tune_interval: int
+            The frequency of tuning. Defaults to 100 iterations.
+        model: PyMC Model
+            Optional model for sampling step. Defaults to None (taken from context).
+        mode:  string or `Mode` instance.
```
Comment (twiecki) [inline pymc3/step_methods/metropolis.py]:
```suggestion
        mode: string or `Mode` instance.
```

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Thanks @ashish-hacker!

**Comment by ashish-hacker** [comment]
Thanks @twiecki for merging. Can I do this for all the classes for all the files?

**Comment by Sayam753** [comment]
If the entire codebase docstrings are to be moved, it will be a good idea to ensure they follow numpy style using pydocstyle and fix them alongside. See #3713

**Comment by ashish-hacker** [comment]
Sure @Sayam753 :) Are you still working on #3713 ?

**Comment by Sayam753** [comment]
No. But will be happy to contribute in these infrastructural issues along with #4354 and #4044.
Also, I am not sure if there are plans to resolve them before or after v4 release.

**Pull request merged**